File: wlr/SPX.Text/TextManager.cs
using System.Collections.Generic;
using System.Linq;
using Microsoft.Xna.Framework.Graphics;
using SPX.Core;

namespace SPX.Text
{
    public class TextType
    {
        public const int Action = 0;
        public const int Inventory = 1;
    }

    public class TextManager
    {
        static private SpriteFont actionFont;
        static private List<Text> m_contents = new List<Text>();

        public static void Add(Text textToAdd)
        {
            if(!m_contents.Contains(textToAdd))
            {
                m_contents.Add(textToAdd);    
            }
        }

        public static void Clear()
        {
            m_contents.Clear();
        }

        static public SpriteFont GetFont()
        {
            return actionFont;
        }

        public static void Update()
        {
            for (int ii = 0; ii < m_contents.Count; ii++)
            {
                if(m_contents[ii].Update()<=0)
                {
                    m_contents.Remove(m_contents[ii]);
                    ii--;
                }
            }
        }

        public static void Draw()
        {
            foreach (Text component in m_contents)
            {
                if (XnaManager.Renderer != null)
                {
                    component.Draw();
                }
            }
        }

        public static void LoadContent()
        {
            actionFont = XnaManager.GetActionFont();
        }
    }
}

File: aigilas/src/sps/text/TextManager.java
package sps.text;

import sps.core.Spx;

import java.util.ArrayList;
import java.util.List;

public class TextManager {
    static private final List<Text> _contents = new ArrayList<Text>();

    public static void add(Text textToAdd) {
        if (!_contents.contains(textToAdd)) {
            _contents.add(textToAdd);
        }
    }

    public static void clear() {
        _contents.clear();
    }

    public static void update() {
        for (int ii = 0; ii < _contents.size(); ii++) {
            if (_contents.get(ii).update() <= 0) {
                _contents.remove(_contents.get(ii));
                ii--;
            }
        }
    }

    public static void draw() {
        for (Text component : _contents) {
            if (Spx.Renderer != null) {
                component.draw();
            }
        }
    }
}
